### PR TITLE
Add support for searching for group of sites

### PR DESCRIPTION
**Pull request by unDemian**
Support for D58442-code discussion.

#### Changes proposed in this Pull Request:

*

#### Testing instructions:

*

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php [hunk 1]
```diff
@@ -161,6 +161,12 @@ protected function inject_javascript_options() {
 			$excluded_post_types = array();
 		}
 
+		$group_id = '';
```
Comment (jsnmoon) [inline projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:164]: Perhaps `null` or `false` would be more appropriate here?

**Review by jsnmoon — COMMENTED** [review]
The overall approach looks reasonable. If the site is part of a Workspace, then individual site search will be permanently replaced by multi-site search. As far as I can see, there's no way to configure this.

Is this correct?

**Comment by unDemian** [comment]
> The overall approach looks reasonable. If the site is part of a Workspace, then individual site search will be permanently replaced by multi-site search. As far as I can see, there's no way to configure this.

That is correct, but now that I think some more about it, we should support single site search even if it is a workspace time.

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php (hunk 1 above)
Comment (unDemian) [reply projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php]: Good point, updated.

**Comment by gibrown** [comment]
> now that I think some more about it, we should support single site search even if it is a workspace time.

You'll probably want to support some sort of search within hub that only searches the FG or the set of users. You could probably just handle that with some extra filtering by blog_id etc even though the search api request goes against the entire group id at all times. Individual links could then open the search modal with the filter already applied (see https://jetpack.com/support/search/install-search-on-your-site/#add-overlay-link) and that also allows the user to quickly expand their search outside of the FG/users very easily without navigating elsewhere.

Summary: I'd recommend always searching against the whole group and adding filters for the different cases.

**Comment by unDemian** [comment]
This was a good conversation starter, will close for now as I will create a new jetpack PR for handling the p2 filtering.

**Pull request closed without merging**